exporter.go

```diff
@@ -106,23 +141,40 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	mapper.mappingsCountMetric.Collect(ch)
 }
 
+func (e *Exporter) retryToDialBeanstalk() (*beanstalk.Conn, error) {
+	for i := 0; i < e.config.numberOfRetry; i++ {
+		c, err := beanstalk.Dial("tcp", e.config.address)
+		if err == nil {
+			return c, nil
+		}
+		e.scrapeConnectionErrorMetric.Inc()
+		log.Errorf("Error. Can't connect to beanstalk: %v retrying", err)
+		time.Sleep(e.config.retrySleepTimer)
+	}
+	return nil, fmt.Errorf(" %d connection retry was failed. skipping current stats", e.config.numberOfRetry)
```

Comment:
Why skipping current stats?

Reply:
Because, we shouldnt block the http server thread. You can understand that there is a problem with beanstalk using `scrapeConnectionErrorMetric` metrics. If you try to continue forever, eventually prometheus client will timeout and each time it is going to create unnecessary sub gorutines. That is why we need to skip for particular time.